pkg/storaged/crypto-keyslots.jsx

```diff
@@ -395,15 +395,25 @@ function ensure_non_root_nbde_support(steps, progress, client, block) {
             .then(() => ensure_crypto_option(steps, progress, client, block, "_netdev"));
 }
 
-function ensure_nbde_support(steps, progress, client, block) {
-    const cleartext = client.blocks_cleartext[block.path];
-    const crypto = client.blocks_crypto[block.path];
-    const fsys_config = cleartext
-        ? cleartext.Configuration.find(c => c[0] == "fstab")
-        : crypto.ChildConfiguration.find(c => c[0] == "fstab");
-    const dir = decode_filename(fsys_config[1].dir.v);
+function contains_rootfs(client, path) {
+    const block = client.blocks[path];
+    const crypto = client.blocks_crypto[path];
+    let fsys_config = null;
 
-    if (dir == "/") {
+    if (block)
+        fsys_config = block.Configuration.find(c => c[0] == "fstab");
+    if (crypto)
+        fsys_config = crypto.ChildConfiguration.find(c => c[0] == "fstab");
```

Comment:
I suppose that works, but it feels a bit brittle and too implicit without an `else` (or possibly reversing the check). Could it ever happen that `block` is in fstab, `crypto` exists, but isn't in fstab?

Reply:
Right, adding a "else" is cleaner.
(When crypto is non-null, then block should never have an entry in fstab, since it contains a LUKS container.)